### PR TITLE
Integrity check is only done if there is WAL or SHM file present

### PR DESCRIPTION
**Pull request by xoriole**
Integrity check at the startup is done only when necessary because it is quite expensive and slows down the startup process. It seems it is sufficient to do integrity check only when there are leftover rollback files (WAL and/or SHM) present.

### CODE REVIEW

**Comment by qstokkink** [comment]
As there is no unit test: did you check this with a naturally created wal and shm file?

**Comment by xoriole** [comment]
Yes, I confirmed with a manual test for it.

**Comment by xoriole** [comment]
retest this please

File: Tribler/Test/Core/test_sqlitecachedb.py [hunk 1]
```diff
@@ -127,6 +127,23 @@ def execute(sql):
         sqlite_test_2.execute = execute
         sqlite_test_2.initialize()
 
+    @blocking_call_on_reactor_thread
+    def test_integrity_check_triggerd(self):
```
Comment (qstokkink) [inline Tribler/Test/Core/test_sqlitecachedb.py]: Please change `triggerd` to `triggered`.

File: Tribler/Test/Core/test_sqlitecachedb.py [hunk 2]
```diff
@@ -127,6 +127,23 @@ def execute(sql):
...
+        def do_integrity_check(expected=False):
+            self.assertTrue(expected, "Integrity check was not expected")
+
```
Comment (qstokkink) [inline Tribler/Test/Core/test_sqlitecachedb.py]: Please set `do_integrity_check.called = False` here and then set `do_integrity_check.called = True` inside `def do_integrity_check()`. Then you can do an assert on whether it was called or not externally.

**Pull request merged**